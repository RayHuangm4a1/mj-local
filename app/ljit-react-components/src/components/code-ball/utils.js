// code ball enums
export const TypeEnums = {
	CIRCLE: 'circle',
	ROUND: 'round',			// 正方形帶圓角
	RECTANGLE: 'rectangle', // 長方形帶圓角
	ANIMAL: 'animal',
};

export const StatusTypeEnum = {
	PRIMARY: 'primary',
	SELECTED: 'selected',
	DISABLED: 'disabled',
};