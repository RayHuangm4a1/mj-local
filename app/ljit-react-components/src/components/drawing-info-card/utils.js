import PropTypes from 'prop-types';

export const drawingInfoCardPropTypes = {
	opencode: PropTypes.string.isRequired,
	size: PropTypes.string,
	fontSize: PropTypes.string,
};

export const drawingInfoCardElementSharingPropTypes = {
	splitOpencodes: PropTypes.arrayOf(PropTypes.string),
	size: PropTypes.string,
	fontSize: PropTypes.string,
};
