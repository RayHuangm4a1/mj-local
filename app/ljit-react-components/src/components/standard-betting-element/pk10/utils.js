export const PK10_TEXT_INPUT_PLACEHOLDER = `说明：1.支持常见的各种单式格式，间隔符如:空格、回车、逗号、分号。
	   2.上传文件必须是.txt或是.csv格式
	   3.文件较大时会导致上传时间较长，请耐心等候
	   格式范例: 01 02 03 04 05,02 03 04 05 06,03 04 05 06 07`;

export function generatePk10FilterData(data) {
	const isLastSpace = /\s/.test(data.charAt(data.length - 1));
	const lastCharacter = isLastSpace ? ' ' : '';
	const dataArray1 = data.split(/[,\n;]/);
	const dataArray2 = dataArray1.map(function (data) {
		const _data = data.trim().split(/[\s]/).filter((item) => !!item);

		return [...new Set(_data),].join(' ');
	});

	const dataSet = new Set(dataArray2);

	return [...dataSet,].join(',') + lastCharacter;
}
