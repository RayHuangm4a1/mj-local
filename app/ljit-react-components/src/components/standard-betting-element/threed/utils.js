export const DEFAULT_THREED_TEXT_INPUT_PLACEHOLDER = `说明：1.支持常见的各种单式格式，间隔符如:空格、回车、逗号、分号。
	   2.上传文件必须是.txt或是.csv格式
	   3.文件较大时会导致上传时间较长，请耐心等候
	   格式范例: 123,456,789
`;

export function generateThreedFilterData(data) {
	const isLastSpace = /\s/.test(data.charAt(data.length - 1));
	const lastCharacter = isLastSpace ? ' ' : '';
	const dataArray1 = data.split(/[,\n; ]/);
	const dataArray2 = dataArray1.map(function (data) {
		const _data = data.trim().split(/[\s]/).filter((item) => !!item);

		return [...new Set(_data),].join(' ');
	});

	const dataSet = new Set(dataArray2);

	return [...dataSet,].join(',') + lastCharacter;
}
