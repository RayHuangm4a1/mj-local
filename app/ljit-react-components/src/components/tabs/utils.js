export const PREFIX_CLASS = 'ljit-tabs';

export const TabPositionEnum = {
	TOP: 'top',
	RIGHT: 'right',
	BOTTOM: 'bottom',
	LEFT: 'left',
};

export const TabTypeEnum = {
	LINE: 'line',
	CARD: 'card',
	CARD_GROUP: 'card-group',
	LIST: 'list',
};

export const TabTypeMap = {
	[TabTypeEnum.LINE]: 'line',
	[TabTypeEnum.CARD]: 'card',
	[TabTypeEnum.CARD_GROUP]: 'card',
	[TabTypeEnum.LIST]: 'list',
};
