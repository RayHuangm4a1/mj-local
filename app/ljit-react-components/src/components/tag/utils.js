export const ColorEnums = {
	WARMORANGE500: 'warmOrange500',
	WARMORANGE700: 'warmOrange700',
	WARMORANGE900: 'warmOrange900',
	GRADIENT_ORANGE: 'gradient-orange',
	ORANGE_RED: 'orange-red',
	LIGHTGREY: 'light-grey',
	BLACK: 'black',
	WHITE: 'white',
	GREEN: 'green',
	RED: 'red',
};

export const ShapeEnums = {
	RECTANGLE: 'rectangle',
	CIRCLE: 'circle',
	ROUND: 'round',
};

export const SizeEnum = {
	//Circle/Round: 48px; Rectangle: 170px
	EXTRA_LARGE: 'extra-large',
	//Circle/Round: 40px; Rectangle: 112px
	LARGE: 'large',
	//Circle/Round: 32px; Rectangle: 64px
	MIDDLE: 'middle',
	//Circle/Round: 24px; Rectangle: 54px
	SMALL: 'small',
	//Circle/Round: 20px; Rectangle: 40px
	EXTRA_SMALL: 'extra-small',
	//100%
	FULL: 'full',
};

export const FontSizeEnum = {
	//32px
	EXTRA_LARGE: 'extra-large',
	//24px
	LARGE: 'large',
	//20px
	MIDDLE: 'middle',
	//14px
	SMALL: 'small',
	//12px
	EXTRA_SMALL: 'extra-small',
};

export const BorderRadiusEnum = {
	WITH_4: 'with-4',
	WITH_12: 'with-12',
	WITH_16: 'with-16',
};