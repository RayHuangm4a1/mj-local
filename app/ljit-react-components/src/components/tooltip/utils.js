import PropTypes from 'prop-types';

export const PlacementEnums = {
	TOP: 'top',
	LEFT: 'left',
	RIGHT: 'right',
	BOTTOM: 'bottom',
	TOP_LEFT: 'topLeft',
	TOP_RIGHT: 'topRight',
	BOTTOM_LEFT: 'bottomLeft',
	BOTTOM_RIGHT: 'bottomRight',
	LEFT_TOP: 'leftTop',
	LEFT_BOTTOM: 'leftBottom',
	RIGHT_TOP: 'rightTop',
	RIGHT_BOTTOM: 'rightBottom',
};
export const TriggerTypeEnums = {
	HOVER: 'hover',
	FOCUS: 'focus',
	CLICK: 'click',
	CONTEXT_MENU: 'contextMenu',
};
export const ColorEnums = {
	DEFAULT: 'default',
	WHITE: 'white',
};

export const TooltipPropTypes = {
	title: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.node,
	]),
	placement: PropTypes.oneOf(Object.values(PlacementEnums)),
	trigger: PropTypes.oneOf(Object.values(TriggerTypeEnums)),
	children: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.node,
	]),
	className: PropTypes.string,
	isArrowPointAtCenter: PropTypes.bool,
	overlayColor: PropTypes.oneOf(Object.values(ColorEnums).concat('')),
	overlayClassName: PropTypes.string,
	overlayStyle: PropTypes.object,
};
