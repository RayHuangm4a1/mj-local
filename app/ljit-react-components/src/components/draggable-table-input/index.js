import React, { Component, } from 'react';
import PropTypes from 'prop-types';
import { Table, Icon, } from 'antd';
import {
	DragDropContext,
} from 'react-beautiful-dnd';
import cx from 'classnames';
import DroppableBody from './droppable-body';
import DragableRow from './draggable-row';
import { PREFIX_CLASS, reorder, } from './utils';
import './style.styl';

const AlignTypeEnums = {
	LEFT: 'left',
	CENTER: 'center',
	RIGHT: 'right',
};
const {
	LEFT,
	CENTER,
	RIGHT,
} = AlignTypeEnums;

const propTypes = {
	// string|Function(record):string
	rowKey: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.func,
	]).isRequired,
	tableName: PropTypes.string.isRequired,
	value: PropTypes.arrayOf(PropTypes.object),
	columns: PropTypes.arrayOf(PropTypes.shape({
		dataIndex: PropTypes.string.isRequired,
		title: PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.node,
		]).isRequired,
	})).isRequired,
	alignType: PropTypes.oneOf([
		LEFT,
		CENTER,
		RIGHT,
	]),
	isDragColumnVisible: PropTypes.bool,
	onChange: PropTypes.func,
	className: PropTypes.string,
};
const defaultProps = {
	value: [],
	columns: [],
	alignType: CENTER,
	isDragColumnVisible: true,
	onChange: () => {},
	className: '',
};

class DraggableTableInput extends Component {
	constructor() {
		super();
		this._getKey = this._getKey.bind(this);
		this._getColumns = this._getColumns.bind(this);
		this._handleSetRowProps = this._handleSetRowProps.bind(this);
		this._handleDragEnd = this._handleDragEnd.bind(this);
		this._renderSubComponents = this._renderSubComponents.bind(this);
		this._wrapperDroppableBody = this._wrapperDroppableBody.bind(this);
	}

	_getKey(record = {}) {
		const {
			rowKey,
		} = this.props;

		if (typeof rowKey === 'function') {
			return rowKey(record);
		}

		return record[rowKey];
	}

	_getColumns() {
		const {
			isDragColumnVisible,
			columns,
		} = this.props;
		let result = columns.slice();

		if (!isDragColumnVisible) {
			return result;
		}

		const dragColumn = {
			dataIndex: 'drag',
			width: 60,
			render: () => <Icon type="menu" />,
		};

		result.unshift(dragColumn);
		return result;
	}

	_handleSetRowProps(record, rowIndex) {
		const {
			tableName,
		} = this.props;
		const key = this._getKey(record);
		const draggableId = `${tableName}-row-${key}`;

		return {
			record,
			rowIndex,
			draggableId,
		};
	}

	_handleDragEnd(result = {}) {
		const {
			value,
			onChange,
		} = this.props;

		if (
			!result.destination
			|| result.destination.index === result.source.index
		) {
			return;
		}

		const data = reorder(
			value,
			result.source.index,
			result.destination.index,
		);

		onChange(data);
	}
	_wrapperDroppableBody(props) {
		const {
			tableName,
		} = this.props;

		return <DroppableBody {...props} droppableId={tableName} />;
	}

	_renderSubComponents() {
		const { _wrapperDroppableBody, } = this;

		return {
			body: {
				wrapper: _wrapperDroppableBody,
				row: DragableRow,
			},
		};
	}

	render() {
		const {
			rowKey,
			alignType,
			tableName,
			className,
			value,
		} = this.props;
		const columns = this._getColumns();

		return (
			<DragDropContext onDragEnd={this._handleDragEnd}>
				<Table
					rowKey={rowKey}
					className={cx(`${PREFIX_CLASS} ${PREFIX_CLASS}--${tableName}`, {
						[`${PREFIX_CLASS}--align-left`]: alignType === LEFT,
						[`${PREFIX_CLASS}--align-center`]: alignType === CENTER,
						[`${PREFIX_CLASS}--align-right`]: alignType === RIGHT,
					}, className)}
					columns={columns}
					dataSource={value}
					pagination={false}
					onRow={this._handleSetRowProps}
					components={this._renderSubComponents()}
				/>
			</DragDropContext>
		);
	}
}

DraggableTableInput.propTypes = propTypes;
DraggableTableInput.defaultProps = defaultProps;

DraggableTableInput.AlignTypeEnums = AlignTypeEnums;

export default DraggableTableInput;
