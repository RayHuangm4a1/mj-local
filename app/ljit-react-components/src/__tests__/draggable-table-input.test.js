import React from 'react';
import { shallow, mount, } from 'enzyme';
import { dragAndDrop, } from '../../helpers';
import { PREFIX_CLASS, reorder, } from '../components/draggable-table-input/utils';

const {
	simulateDragAndDrop,
} = dragAndDrop;

describe('DraggableTableInput', () => {
	let DraggableTableInput;
	let tableName;
	let rowKey;
	let dataColumns;
	let dataSource;

	beforeEach(() => {
		DraggableTableInput = require('../components/draggable-table-input').default;

		tableName = 'orders';
		rowKey = '_id';

		dataColumns = [
			{
				title: 'Title',
				dataIndex: 'title',
			},
			{
				title: 'Order',
				dataIndex: 'order',
			},
		];
		dataSource = [
			{
				title: 'title 1',
				order: 1,
				_id: '0',
			},
			{
				title: 'title 2',
				order: 2,
				_id: '1',
			},
			{
				title: 'title 3',
				order: 3,
				_id: '2',
			},
		];
	});
	afterEach(() => {
		tableName = undefined;
		rowKey = undefined;
		DraggableTableInput = undefined;
		dataColumns = undefined;
		dataSource = undefined;
	});

	it('should handle default props', () => {
		const {
			value,
			columns,
			alignType,
			isDragColumnVisible,
			onChange,
			className,
		} = DraggableTableInput.defaultProps;

		expect(columns).toMatchObject([]);
		expect(value).toMatchObject([]);
		expect(alignType).toEqual('center');
		expect(className).toEqual('');
		expect(isDragColumnVisible).toEqual(true);
		expect(onChange).toBeDefined();
		expect(onChange).toBeInstanceOf(Function);
	});

	it('should renders correctly', () => {
		const className = 'mock-class';
		const wrapper = shallow(
			<DraggableTableInput
				tableName={tableName}
				rowKey={rowKey}
				value={dataSource}
				columns={dataColumns}
				className={className}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-draggable-table-input', () => {
		const wrapper = shallow(
			<DraggableTableInput
				tableName={tableName}
				rowKey={rowKey}
				columns={dataColumns}
			/>
		);
		const expected = 1;

		expect(wrapper.find('.ljit-draggable-table-input')).toHaveLength(expected);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(
			<DraggableTableInput
				tableName={tableName}
				rowKey={rowKey}
				className={className}
				columns={dataColumns}
			/>
		);
		const expected = 1;

		expect(wrapper.find(`.${className}`)).toHaveLength(expected);
	});

	it('should mount in a full DOM', () => {
		const alignType = 'right';
		const className = 'mock-class';
		const isDragColumnVisible = true;
		const onDragEnd = () => {};
		const wrapper = mount(
			<DraggableTableInput
				tableName={tableName}
				rowKey={rowKey}
				value={dataSource}
				columns={dataColumns}
				alignType={alignType}
				className={className}
				isDragColumnVisible={isDragColumnVisible}
				onDragEnd={onDragEnd}
			/>
		);

		expect(wrapper.props().tableName).toEqual(tableName);
		expect(wrapper.props().rowKey).toEqual(rowKey);
		expect(wrapper.props().value).toEqual(dataSource);
		expect(wrapper.props().columns).toEqual(dataColumns);
		expect(wrapper.props().alignType).toEqual(alignType);
		expect(wrapper.props().className).toEqual(className);
		expect(wrapper.props().isDragColumnVisible).toEqual(isDragColumnVisible);
		expect(wrapper.props().onDragEnd).toEqual(onDragEnd);
	});

	it('should handle onChange', () => {
		const onChange = jest.fn();
		const wrapper = mount(
			<DraggableTableInput
				tableName={tableName}
				rowKey={rowKey}
				value={dataSource}
				columns={dataColumns}
				onChange={onChange}
			/>
		);

		simulateDragAndDrop(wrapper, tableName, 1, tableName, 0);

		expect(onChange).toHaveBeenCalled();
	});

	describe('AlignTypeEnums', () => {
		it('should contain left property', () => {
			expect(DraggableTableInput.AlignTypeEnums).toHaveProperty('LEFT', 'left');
		});

		it('should contain center property', () => {
			expect(DraggableTableInput.AlignTypeEnums).toHaveProperty('CENTER', 'center');
		});

		it('should contain right property', () => {
			expect(DraggableTableInput.AlignTypeEnums).toHaveProperty('RIGHT', 'right');
		});
	});

	describe('Whan pass alignType as left, center, or right', () => {
		it('should be selected by class ljit-draggable-table-input--align-left', () => {
			const wrapper = shallow(
				<DraggableTableInput
					tableName={tableName}
					rowKey={rowKey}
					value={dataSource}
					columns={dataColumns}
					alignType="left"
				/>
			);
			const wrappedTable = wrapper.find('.ljit-draggable-table-input').at(0);

			expect(wrappedTable.hasClass('ljit-draggable-table-input--align-left')).toEqual(true);
		});

		it('should be selected by class ljit-draggable-table-input--align-center', () => {
			const wrapper = shallow(
				<DraggableTableInput
					tableName={tableName}
					rowKey={rowKey}
					value={dataSource}
					columns={dataColumns}
					alignType="center"
				/>
			);
			const wrappedTable = wrapper.find('.ljit-draggable-table-input').at(0);

			expect(wrappedTable.hasClass('ljit-draggable-table-input--align-center')).toEqual(true);
		});

		it('should be selected by class ljit-draggable-table-input--align-right', () => {
			const wrapper = shallow(
				<DraggableTableInput
					tableName={tableName}
					rowKey={rowKey}
					value={dataSource}
					columns={dataColumns}
					alignType="right"
				/>
			);
			const wrappedTable = wrapper.find('.ljit-draggable-table-input').at(0);

			expect(wrappedTable.hasClass('ljit-draggable-table-input--align-right')).toEqual(true);
		});
	});

	describe('utils', () => {
		describe('PREFIX_CLASS', () => {
			it('should PREFIX_CLASS to equal ljit-draggable-table-input', () => {
				const expected = 'ljit-draggable-table-input';

				expect(PREFIX_CLASS).toEqual(expected);
			});
		});

		describe('reorder', () => {
			it('should remove dataSource at 0, then insert to 1', () => {
				const startIndex = 0;
				const endIndex = 1;
				const received = reorder(dataSource, startIndex, endIndex);
				const expected = [
					{
						title: 'title 2',
						order: 2,
						_id: '1',
					},
					{
						title: 'title 1',
						order: 1,
						_id: '0',
					},
					{
						title: 'title 3',
						order: 3,
						_id: '2',
					},
				];

				expect(received).toMatchObject(expected);
			});

			it('should remove dataSource at 0, then insert to 2', () => {
				const startIndex = 0;
				const endIndex = 2;
				const received = reorder(dataSource, startIndex, endIndex);
				const expected = [
					{
						title: 'title 2',
						order: 2,
						_id: '1',
					},
					{
						title: 'title 3',
						order: 3,
						_id: '2',
					},
					{
						title: 'title 1',
						order: 1,
						_id: '0',
					},
				];

				expect(received).toMatchObject(expected);
			});
		});
	});
});
