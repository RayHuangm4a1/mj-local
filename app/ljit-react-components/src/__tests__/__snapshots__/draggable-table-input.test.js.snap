// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableTableInput should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DraggableTableInput
    alignType="center"
    className="mock-class"
    columns={
      Array [
        Object {
          "dataIndex": "title",
          "title": "Title",
        },
        Object {
          "dataIndex": "order",
          "title": "Order",
        },
      ]
    }
    isDragColumnVisible={true}
    onChange={[Function]}
    rowKey="_id"
    tableName="orders"
    value={
      Array [
        Object {
          "_id": "0",
          "order": 1,
          "title": "title 1",
        },
        Object {
          "_id": "1",
          "order": 2,
          "title": "title 2",
        },
        Object {
          "_id": "2",
          "order": 3,
          "title": "title 3",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Table
        bordered={false}
        className="ljit-draggable-table-input ljit-draggable-table-input--orders ljit-draggable-table-input--align-center mock-class"
        columns={
          Array [
            Object {
              "dataIndex": "drag",
              "render": [Function],
              "width": 60,
            },
            Object {
              "dataIndex": "title",
              "title": "Title",
            },
            Object {
              "dataIndex": "order",
              "title": "Order",
            },
          ]
        }
        components={
          Object {
            "body": Object {
              "row": [Function],
              "wrapper": [Function],
            },
          }
        }
        dataSource={
          Array [
            Object {
              "_id": "0",
              "order": 1,
              "title": "title 1",
            },
            Object {
              "_id": "1",
              "order": 2,
              "title": "title 2",
            },
            Object {
              "_id": "2",
              "order": 3,
              "title": "title 3",
            },
          ]
        }
        indentSize={20}
        loading={false}
        locale={Object {}}
        onRow={[Function]}
        pagination={false}
        rowKey="_id"
        showHeader={true}
        size="default"
        sortDirections={
          Array [
            "ascend",
            "descend",
          ]
        }
        useFixedHeader={false}
      />,
      "onDragEnd": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bordered": false,
        "className": "ljit-draggable-table-input ljit-draggable-table-input--orders ljit-draggable-table-input--align-center mock-class",
        "columns": Array [
          Object {
            "dataIndex": "drag",
            "render": [Function],
            "width": 60,
          },
          Object {
            "dataIndex": "title",
            "title": "Title",
          },
          Object {
            "dataIndex": "order",
            "title": "Order",
          },
        ],
        "components": Object {
          "body": Object {
            "row": [Function],
            "wrapper": [Function],
          },
        },
        "dataSource": Array [
          Object {
            "_id": "0",
            "order": 1,
            "title": "title 1",
          },
          Object {
            "_id": "1",
            "order": 2,
            "title": "title 2",
          },
          Object {
            "_id": "2",
            "order": 3,
            "title": "title 3",
          },
        ],
        "indentSize": 20,
        "loading": false,
        "locale": Object {},
        "onRow": [Function],
        "pagination": false,
        "rowKey": "_id",
        "showHeader": true,
        "size": "default",
        "sortDirections": Array [
          "ascend",
          "descend",
        ],
        "useFixedHeader": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Table
          bordered={false}
          className="ljit-draggable-table-input ljit-draggable-table-input--orders ljit-draggable-table-input--align-center mock-class"
          columns={
            Array [
              Object {
                "dataIndex": "drag",
                "render": [Function],
                "width": 60,
              },
              Object {
                "dataIndex": "title",
                "title": "Title",
              },
              Object {
                "dataIndex": "order",
                "title": "Order",
              },
            ]
          }
          components={
            Object {
              "body": Object {
                "row": [Function],
                "wrapper": [Function],
              },
            }
          }
          dataSource={
            Array [
              Object {
                "_id": "0",
                "order": 1,
                "title": "title 1",
              },
              Object {
                "_id": "1",
                "order": 2,
                "title": "title 2",
              },
              Object {
                "_id": "2",
                "order": 3,
                "title": "title 3",
              },
            ]
          }
          indentSize={20}
          loading={false}
          locale={Object {}}
          onRow={[Function]}
          pagination={false}
          rowKey="_id"
          showHeader={true}
          size="default"
          sortDirections={
            Array [
              "ascend",
              "descend",
            ]
          }
          useFixedHeader={false}
        />,
        "onDragEnd": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bordered": false,
          "className": "ljit-draggable-table-input ljit-draggable-table-input--orders ljit-draggable-table-input--align-center mock-class",
          "columns": Array [
            Object {
              "dataIndex": "drag",
              "render": [Function],
              "width": 60,
            },
            Object {
              "dataIndex": "title",
              "title": "Title",
            },
            Object {
              "dataIndex": "order",
              "title": "Order",
            },
          ],
          "components": Object {
            "body": Object {
              "row": [Function],
              "wrapper": [Function],
            },
          },
          "dataSource": Array [
            Object {
              "_id": "0",
              "order": 1,
              "title": "title 1",
            },
            Object {
              "_id": "1",
              "order": 2,
              "title": "title 2",
            },
            Object {
              "_id": "2",
              "order": 3,
              "title": "title 3",
            },
          ],
          "indentSize": 20,
          "loading": false,
          "locale": Object {},
          "onRow": [Function],
          "pagination": false,
          "rowKey": "_id",
          "showHeader": true,
          "size": "default",
          "sortDirections": Array [
            "ascend",
            "descend",
          ],
          "useFixedHeader": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
